src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,339 @@
+    support_class = clr.AddReference(support_path)
+
+    import PrincetonInstruments.LightField as lf
+
+    clr.AddReference("System.Collections")
+    clr.AddReference("System.IO")
+    from System.Collections.Generic import List
+    from System import String
+    from System.IO import FileAccess
+except Exception as e:
+    print(f"Exception occurred during import: {e}")
+
+
+class LightfieldApp:
+    def __init__(self, visible):
+
+        self._addinbase = lf.AddIns.AddInBase()
+        self._automation = lf.Automation.Automation(visible, List[String]())
+        self._application = self._automation.LightFieldApplication
+        self._experiment = self._application.Experiment
+
+    @property
+    def automation(self):
+        return self._automation
+
+    @property
+    def addinbase(self):
+        return self._addinbase
+
+    @property
+    def application(self):
+        return self._application
+
+    @property
+    def experiment(self):
+        return self._experiment
+
+    def __del__(self):
+        """
+        Uses Python garbage collection method used to terminate
+        AddInProcess.exe, if this is not done, LightField will not reopen
+        properly.
+        """
+        self.automation.Dispose()
+        print('Closed AddInProcess.exe')
+
+    def set(self, setting, value):
+        """
+        Helper function for setting experiment parameters with basic value
+        checking.
+        """
+
+        if self.experiment.Exists(setting):
+            if self.experiment.IsValid(setting, value):
+
+                self.experiment.SetValue(setting, value)
+
+            else:
+
+                # TODO: might need to add a proper exception
+                print('Invalid value: {} for setting: {}.'.format(value, setting))
+
+        else:
+
+            # TODO: might need to add a proper exception
+            print('Invalid setting:{}'.format(setting))
+
+        return
+
+    def get(self, setting):
+        """
+        Helper function for getting experiment parameters with a basic check
+        to see if a specific setting is valid.
+        """
+
+        if self.experiment.Exists(setting):
+
+            value = self.experiment.GetValue(setting)
+
+        else:
+
+            value = []
+            print('Invalid setting: {}'.format(setting))
+
+        return value
+
+    def load_experiment(self, value):
+        self.experiment.Load(value)
+
+    def acquire(self):
+        """
+        Helper function to acquire the data from the PIXIS-style cameras.
+        Need to check array reorganization since there is only a single vertical pixel.
+        For a single frame, the data is returned as a 2D array (just raw counts
+        from each pixel). Taking a section in the vertical (400-pixel) direction
+        corresponds to wavelength averaging.
+        For multiple frames, each exposure is stacked in a third dimension, so
+        averaging can be performed simply by summing over the different planes.
```

Comment:
Kudos on including a long Docstring! :) 

But I wonder if this Docstring can be more clear.  Are these notes for the developer (you) or is this information that as a user of this object.function I need to understand?

Reply:
Originally for me as I was having a hard time understanding what I was trying to do with this function, but I can make it  clearer.